web/client-api/src/main/java/io/deephaven/web/client/api/widget/plot/JsAxisDescriptor.java

```diff
@@ -12,8 +12,11 @@
 
 @JsType(name = "AxisDescriptor", namespace = "dh.plot")
 public class JsAxisDescriptor {
+    // TODO (deephaven-core#3442) change to some kind of String+int union type
     public String formatType;
+    // TODO (deephaven-core#3442) change to some kind of String+int union type
     public String type;
+    // TODO (deephaven-core#3442) change to some kind of String+int union type
     public String position;
 
     public boolean log = false;
```

Comment:
@mattrunyon and I had this chat in slack (you were in the channel too, but I don't recall if you weighed in) - we have 4 cases to handle, potentially:
 * not optional, not nullable - required, non-null values
 * optional and nullable - not required, but if you specify it, it cannot be null
 * not optional, but nullable - required, but you can explicitly state null
 * optional, but not nullable - not required, and you can explicitly state null
 
 (note that I'm using "null" as a placeholder for "null or undefined")

If we need all four states, we must have both `@JsNullable` and a `@TsOptional` that we can decorate on properties to make this clear. On the other hand, if we really only need the first and last, then in terms of values that are possible, adding `@JsNullable` to a property means the following:
 * it can be a real value
 * it can be assigned to null
 * it can be _assigned to undefined_
 * it can be left undefined

The last three are all the same in Java, and JS insists on a distinction between the last two. Matt and I decided that the last two _really_ doesnt matter, as long as you aren't asking the object directly if it has the property defined on it, but just reading or writing to the property.

But null vs undefined still matters - if it is allowed to be null but not undefined, then TS will tell you that a `foo === null` check is correct, when you actually want a `foo == null` check to normalize out the undefined value - and vice versa. Assuming we support "nullable properties are automatically optional" we can drop the `undefined` from the union, but we can't/shouldn't drop the null, or nulls could escape.

For Java/GWT's part, there is no difference between a field that hasn't been initialized and one that has been initialized to null, so anywhere we handle a null, we have to have some kind of undefined handling too.